fdbserver/DataDistributionQueue.actor.cpp

```diff
@@ -1307,14 +1363,59 @@ ACTOR Future<Void> dataDistributionRelocator(DDQueueData* self,
 
 						bestTeams.emplace_back(bestTeam.first.get(), bestTeam.second);
 					}
+
+					// update physicalShardIDCandidate
+					if (CLIENT_KNOBS->SHARD_ENCODE_LOCATION_METADATA && tciIndex==0 && rd.dataMoveIdSetDelayed) { // && rd.dataMoveIdSetDelayed && foundTeams
```

Comment:
I feel this applies only to the case `!rd.isRestore()`, since when we are restoring a data move, the dest teams are determined, and we should not change them.

Reply:
For the case rd.keys != ranges[r] && rd.isRestore(), we don't have rrs.dataMoveId decided yet. Seems we need decide a dataMoveId for this case here.